io/src/main/java/io/smallrye/common/io/FileUtils.java

```diff
@@ -0,0 +1,565 @@
+
+    /**
+     * A fluent style read access check, returning the passed argument if path is a
+     * directory. Check is done exactly as
+     * {@link java.nio.file.Files#isReadable(Path)} but with explicit null check and
+     * unchecked {@code java.io.UncheckedIOException}.
+     *
+     * @param path to check, must be not null
+     * @return the passed path if read access is possible
+     * @throws IllegalArgumentException if the path is {@code null}
+     * @throws UncheckedIOException if no read access or determination of access
+     *         rights fails
+     */
+    public static File checkIsReadable(final File file) throws UncheckedIOException {
+        checkNotNullParam("file", file);
+        return PathUtils.doCheckAccess(file.toPath(), AccessMode.READ).toFile();
+    }
+
+    /**
+     * A fluent style write access check, returning the passed argument if path is a
+     * directory. Check is done exactly as
+     * {@link java.nio.file.Files#isWritable(Path)} but with explicit null check and
+     * unchecked {@code java.io.UncheckedIOException}.
+     *
+     * @param path to check, must be not null
+     * @return the passed path if write access is possible
+     * @throws IllegalArgumentException if the path is {@code null}
+     * @throws UncheckedIOException if no write access or determination of
+     *         access rights fails
+     */
+    public static File checkIsWritable(final File file) {
+        checkNotNullParam("file", file);
+        return PathUtils.doCheckAccess(file.toPath(), AccessMode.WRITE).toFile();
+    }
+
+    /**
+     * A fluent style execute access check, returning the passed argument if path is
+     * a directory. Check is done exactly as
+     * {@link java.nio.file.Files#isExecutable(Path)} but with explicit null check
+     * and unchecked {@code java.io.UncheckedIOException}.
+     *
+     * @param path to check, must be not null
+     * @return the passed path if execute access is possible
+     * @throws IllegalArgumentException if the path is {@code null}
+     * @throws UncheckedIOException if no execute access or determination of
+     *         access rights fails
+     */
+    public static File checkIsExecutable(final File file) {
+        checkNotNullParam("file", file);
+        return PathUtils.doCheckAccess(file.toPath(), AccessMode.EXECUTE).toFile();
+    }
+
+    /**
+     * Checks if a directory contains no elements.
+     *
+     * @param dir to check, must not be null
+     * @return the passed dir if it is empty
+     * @throws IllegalArgumentException if the dir is {@code null}
+     * @throws UncheckedIOException if directory contains elements or check
+     *         fails
+     */
+    public static File checkIsEmptyDirectory(final File dir) throws UncheckedIOException {
+        checkNotNullParam("dir", dir);
+        return PathUtils.doCheckIsEmptyDirectory(dir.toPath()).toFile();
+    }
+
+    private FileUtils() {
+        throw Assert.unreachableCode();
```

Comment:
It is enough to have the constructor be private IMO.  This is consistent with other classes in the project I believe.

Reply:
Removed.